frontend/src/lib/k8s/index.ts

```diff
@@ -173,6 +173,86 @@ export function useConnectApi(...apiCalls: (() => CancellablePromise)[]) {
   );
 }
 
+// Label selector examples: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#list-and-watch-filtering
+// deployment selector example: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements
+// Possible operators: https://github.com/kubernetes/apimachinery/blob/be3a79b26814a8d7637d70f4d434a4626ee1c1e7/pkg/selection/operator.go#L24
+// Format rule for expressions: https://github.com/kubernetes/apimachinery/blob/be3a79b26814a8d7637d70f4d434a4626ee1c1e7/pkg/labels/selector.go#L305
+export function labelSelectorToQuery(labelSelector: LabelSelector) {
+  const segments: string[] = [];
+
+  const matchLabels = labelSelector.matchLabels ?? {};
+  for (const k in matchLabels) {
+    segments.push(`${k}=${matchLabels[k]}`);
+  }
+
+  const matchExpressions = labelSelector.matchExpressions ?? [];
+  for (const expr of matchExpressions) {
+    let segment = '';
+    if (expr.operator === 'DoesNotExist') {
+      segment += '!';
+    }
+
+    let needsParensWrap = false;
+    let expectedValuesLength = -1; // -1 means no limit
+
+    segment += expr.key;
+    switch (expr.operator) {
+      case 'Equals':
+        segment += '=';
+        expectedValuesLength = 1;
+        break;
+      case 'DoubleEquals':
+        segment += '==';
+        expectedValuesLength = 1;
+        break;
+      case 'NotEquals':
+        segment += '!=';
+        expectedValuesLength = 1;
+        break;
+      case 'In':
+        segment += ' in ';
+        needsParensWrap = true;
+        break;
+      case 'NotIn':
+        segment += ' notin ';
+        needsParensWrap = true;
+        break;
+      case 'GreaterThan':
+        segment += '>';
```

Comment:
I think these GreaterThan and LessThan are wrong. See https://github.com/kubernetes/apimachinery/blob/be3a79b26814a8d7637d70f4d434a4626ee1c1e7/pkg/selection/operator.go#L31

Reply:
I did some experiment. `<` and `>` are ok. `lt` and `gt` are not allowed in our kubectl.

Reply:
This seems to be some sort of internal label, and ">" and "<" works in testing. I'll mark this as resolved.